workflows/sampletracking.nf

```diff
@@ -40,12 +43,78 @@ workflow SAMPLETRACKING {
     def ch_multiqc_files = Channel.empty()
     def ch_pool_multiqc_files = Channel.empty()
 
+
+    def (ch_sample, ch_snp, ch_rest) = ch_samplesheet
+        .multiMap {meta, sample_bam, sample_bam_index, snp_fastq, snp_bam, snp_bam_index ->
+            sample: [meta, (sample_bam instanceof List && sample_bam.isEmpty() ? null : sample_bam), (sample_bam_index ?: null)]
+            snp   : [meta, (snp_bam   instanceof List && snp_bam.isEmpty()   ? null : snp_bam),     (snp_bam_index   ?: null)]
+            rest  : [meta, (snp_fastq instanceof List && snp_fastq.isEmpty() ? null : snp_fastq)]
```

Comment:
```suggestion
            sample: [meta, sample_bam, sample_bam_index]
            snp   : [meta, snp_bam, snp_bam_index]
            rest  : [meta, snp_fastq]
```

These checks are nice but not really needed. `[]` is considered as an empty entry while still being compatible with process inputs. `null` will cause issues when given to a process.